play-ahc-ws-standalone/src/main/scala/play/api/libs/ws/ahc/CookieBuilder.scala

```diff
@@ -1,20 +1,26 @@
 package play.api.libs.ws.ahc
 
 import play.api.libs.ws.{ DefaultWSCookie, WSCookie }
-import play.shaded.ahc.io.netty.handler.codec.http.HttpHeaders.Names._
-import play.shaded.ahc.org.asynchttpclient.cookie.{ Cookie, CookieDecoder }
+import play.shaded.ahc.io.netty.handler.codec.http.HttpHeaderNames._
+import play.shaded.ahc.io.netty.handler.codec.http.cookie.{ ClientCookieDecoder, Cookie, DefaultCookie }
 
 trait CookieBuilder extends WSCookieConverter {
   def buildCookies(headers: Map[String, Seq[String]]): Seq[WSCookie] = {
-    val option = headers.get(SET_COOKIE2).orElse(headers.get(SET_COOKIE))
+    val option = headers.get(SET_COOKIE2.toString).orElse(headers.get(SET_COOKIE.toString))
     option.map { cookiesHeaders =>
       for {
         value <- cookiesHeaders
-        Some(c) = Option(CookieDecoder.decode(value))
+        Some(c) = Some {
+          useLaxCookieEncoder match {
```

Comment:
nitpick: `if`/`else` would be more idiomatic here.

Reply:
done